### ISSUE TITLE
Every java source file is duplicated in the 3.1.0 source jar in the central maven repository

### ISSUE THREAD

**Issue by akiraly** [reactor/reactor-core#887]
### Expected behavior
The java source files should not be duplicated

### Actual behavior
They are duplicated

### Steps to reproduce
Download file from http://repo1.maven.org/maven2/io/projectreactor/reactor-core/3.1.0.RELEASE/reactor-core-3.1.0.RELEASE-sources.jar
Check contents of the jar.

### Reactor Core version
3.1.0.RELEASE

### JVM version (e.g. `java -version`)
N/A

### OS version (e.g. `uname -a`)
N/A

**Comment by simonbasle** [comment]
I just did that and I don't see any duplication in the source jar:
![image](https://user-images.githubusercontent.com/6986166/30955234-304dca74-a433-11e7-8832-356b207aded0.png)

**Comment by akiraly** [comment]
I checked it on windows with Total Commander and it does show the java files duplicated for me inside the 3.1.0 source jar.
I have checked 3.0.7 source jar as well. That does not show duplication.

Also the 3.0.7 source jar is less than half the size of the 3.1.0 source jar (600 KB vs 1.3 MB).

**Comment by akarnokd** [comment]
I see the duplication:

![image](https://user-images.githubusercontent.com/1269832/30955430-eb91289e-a433-11e7-83dc-7a31d1409e45.png)

Also this Java 9 program shows there are duplicate entries:

```java
        try (JarFile jar = new JarFile("reactor-core-3.1.0.RELEASE-sources.jar")) {
            jar.stream()
                    .collect(Collectors.groupingBy(e -> e.getName()))
            .entrySet()
            .stream()
            .filter(e -> e.getValue().size() > 1)
            .forEach(System.out::println);
        }
```

**Comment by simonbasle** [comment]
Thanks. Upon extraction or listing on the command line with `unzip -l reactor-core-3.1.0.RELEASE-sources.jar` hides the duplication, but if I do `unzip reactor-core-3.1.0.RELEASE-sources.jar -d ./temp` it asks if I want to replace an existing file so there is duplication indeed :(

investigating...

@akiraly has it caused any issue for you eg. in the IDE so far?

**Comment by akiraly** [comment]
It did not cause any issues for me so far so IMHO this is just a minor issue.

**Comment by akarnokd** [comment]
Could be Gradle 4.2 or a config problem with the subprojects:

https://stackoverflow.com/questions/34996571/gradle-multi-project-builds-with-duplicate-source-files

**Comment by sdeleuze** [comment]
Seems to be https://youtrack.jetbrains.com/issue/KT-17564

**Comment by simonbasle** [comment]
Looks like this is indeed the Kotlin plugin issue, and a workaround is to explicitly exclude duplicates (`duplicatesStrategy = DuplicatesStrategy.EXCLUDE`) in the jar task...

@akiraly @akarnokd @sdeleuze thanks for checking the jar with various tools 👍 (tested integration in IntelliJ 2017.2 and Eclipse Neon as well)

So far it doesn't look like it is causing any critical side effect 😌 

I'll list this issue as a know "bug" of the 3.1.0 RELEASE in the release notes and also link to it in the announcement blog I'm preparing.

I'll also keep it open for a week or so, in order to give a chance for people to comment if they encounter any major usability issue as a result of this situation (you never know).

**Comment by sdeleuze** [comment]
This issue has been fixed, the fix will be available in Kotlin Gradle plugin 1.1.60 https://youtrack.jetbrains.com/issue/KT-17564

**Comment by simonbasle** [comment]
closing this as `3.1.1.RELEASE` was released without duplicate sources thanks to the workaround. #915 will track the upgrade of gradle plugin when it becomes available, as well as the removal of the workaround.
